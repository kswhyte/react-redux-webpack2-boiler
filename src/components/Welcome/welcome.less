
.start-call-banner-block {
    padding: 10%;
    .error-container {
        width: 100%;
        height: 100%;
        text-align: center;
        padding-top: 200px;
        color: #e37342;
    }
}

.new-session-button {
    font-size: 1.5rem;
    background: #25a2e1;
    color: white;
}