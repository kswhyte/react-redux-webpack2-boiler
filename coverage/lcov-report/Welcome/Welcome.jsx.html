<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for Welcome/Welcome.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">Welcome</a> Welcome.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>3/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>3/3</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import PropTypes from 'prop-types';
import HgRow from '@hg/three-ui/HgRow';
import HgButton from '@hg/three-ui/HgButton';
&nbsp;
import './welcome.less';
&nbsp;
const Welcome = (props) =&gt; {
    return (
        &lt;div className="start-call-banner-block"&gt;
            &lt;HgRow&gt;
                &lt;h1&gt;Welcome&lt;/h1&gt;
                &lt;hr /&gt;
                &lt;HgButton className="new-session-button" tabIndex={1} onClick={props.createCall} text="Begin New Session" /&gt;
                &lt;div className="error-container"&gt;
                    {
                        (props.showError) &amp;&amp; (
                            &lt;h3&gt;Something went wrong.&lt;/h3&gt;
                        )
                    }
                    {
                        (props.startCallTimeout) &amp;&amp; (
                            &lt;h3&gt;Request timed out.&lt;/h3&gt;
                        )
                    }
                &lt;/div&gt;
            &lt;/HgRow&gt;
        &lt;/div&gt;
    )
}
&nbsp;
Welcome.PropTypes = {
    agent: PropTypes.object,
    createCall: PropTypes.func,
    showError: PropTypes.func,
    startCallTimeout: PropTypes.func
}
&nbsp;
export default Welcome;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Jul 17 2017 10:00:34 GMT-0600 (MDT)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
