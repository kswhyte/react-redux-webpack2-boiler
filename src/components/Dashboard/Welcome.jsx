import React from 'react';
import PropTypes from 'prop-types';
import HgRow from '@hg/three-ui/HgRow';
import HgButton from '@hg/three-ui/HgButton';

import './welcome.css';

const Welcome = props => {
  return (
    <div className="start-call-banner-block">
      <HgRow className="no-margin">
        <h1>Welcome</h1>
        <hr />
        <HgButton
          className="new-session-button"
          tabIndex={1}
          onClick={props.startSessionClick}
          text="Begin New Session"
        />
        <div className="error-container">
          {(props.startError === 'ERROR') && (<h3>Something went wrong.</h3>)}
          {(props.startError === 'TIMEOUT') && (<h3>Request timed out.</h3>)}
        </div>
      </HgRow>
    </div>
  );
};

Welcome.propTypes = {
    startSessionClick: PropTypes.func,
    startError: PropTypes.bool
};

export default Welcome;
