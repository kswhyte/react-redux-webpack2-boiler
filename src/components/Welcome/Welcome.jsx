import React from 'react';
import PropTypes from 'prop-types';
import HgRow from '@hg/three-ui/HgRow';
import HgButton from '@hg/three-ui/HgButton';

import './welcome.css';

const Welcome = (props) => {
    return (
        <div className="start-call-banner-block">
            <HgRow>
                <h1>Welcome</h1>
                <hr />
                <HgButton className="new-session-button" tabIndex={1} onClick={props.createCall} text="Begin New Session" />
                <div className="error-container">
                    {
                        (props.showError) && (
                            <h3>Something went wrong.</h3>
                        )
                    }
                    {
                        (props.startCallTimeout) && (
                            <h3>Request timed out.</h3>
                        )
                    }
                </div>
            </HgRow>
        </div>
    )
}

Welcome.propTypes = {
    createCall: PropTypes.func,
    showError: PropTypes.func,
    startCallTimeout: PropTypes.func
}

export default Welcome;