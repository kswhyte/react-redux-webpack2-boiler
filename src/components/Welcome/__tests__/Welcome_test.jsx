import React from 'react';
// import { storiesOf, action, linkTo } from '@kadira/storybook';
import Welcome from '../Welcome';

// import { specs, describe, it } from 'storybook-addon-specifications';

import {storiesOf, describe, it, specs
} from "../../../../.storybook/facade";

import {shallow, mount} from 'enzyme';
import expect from 'expect';

const createCall = () => {
    console.log("Beginning the call");
}

const startProps = {
    createCall,
    showError: false,
    startCallTimeout: false
}

const errorProps = {
    createCall,
    showError: true,
    startCallTimeout: false
}

const timeoutProps = {
    createCall,
    showError: false,
    startCallTimeout: true
}

const stories = storiesOf('Welcome', module);

stories.add('Welcome page main story', () => {
    const welcomeStory = (<Welcome {...startProps} />);
    specs(() => describe('Show the main start call page', () => {
        it('Show the main start call page', () => {
            let output = mount(welcomeStory);
            expect(output.find('.start-call-banner-block').length).toEqual(1);
            expect(output.find('.new-session-button').text()).toContain('Begin New Session');
            expect(output.find('.start-call-banner-block').text()).toContain('Welcome');
        });
    }));
    return welcomeStory;
});

stories.add('Welcome page main error story', () => {
    const welcomeStory = (<Welcome {...errorProps} />);
    specs(() => describe('Show a generic somethine went wrong error', () => {
        it('Show a generic somethine went wrong error', () => {
            let output = mount(welcomeStory);
            expect(output.find('h3').text()).toContain('Something went wrong');
        });
    }));
    return welcomeStory;
});

stories.add('Welcome page timeout error story', () => {
    const welcomeStory = (<Welcome {...timeoutProps} />);
    specs(() => describe('Show a timeout error', () => {
        it('Show a timeout error', () => {
            let output = mount(welcomeStory);
            expect(output.find('h3').text()).toContain('Request timed out');
        });
    }));
    return welcomeStory;
});