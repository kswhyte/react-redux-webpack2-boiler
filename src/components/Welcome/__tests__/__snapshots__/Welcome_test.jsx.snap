// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome page main error story 1`] = `
<div
  className="start-call-banner-block"
>
  <div>
    <h1>
      Welcome
    </h1>
    <hr />
    <div>
      
    </div>
    <div
      className="error-container"
    >
      <h3>
        Something went wrong.
      </h3>
    </div>
  </div>
</div>
`;

exports[`Welcome page main story 1`] = `
<div
  className="start-call-banner-block"
>
  <div>
    <h1>
      Welcome
    </h1>
    <hr />
    <div>
      
    </div>
    <div
      className="error-container"
    />
  </div>
</div>
`;

exports[`Welcome page timeout error story 1`] = `
<div
  className="start-call-banner-block"
>
  <div>
    <h1>
      Welcome
    </h1>
    <hr />
    <div>
      
    </div>
    <div
      className="error-container"
    >
      <h3>
        Request timed out.
      </h3>
    </div>
  </div>
</div>
`;
